lib/actions/purgeTable.js

```diff
@@ -7,9 +7,30 @@ const findPrimaryKey = context => {
 
   return context.dynamodb.describeTable(params).promise()
     .then(tableDescription => {
-      context.primaryKey = tableDescription.Table.KeySchema.find(
-        element => element.KeyType === 'HASH').AttributeName
-      return context
+      const keys = {};
+
+      let primaryHash = null;
+      let primaryRange = null;
+
+      try {
+        if(primaryHash = tableDescription.Table.KeySchema.find(element => element.KeyType === 'HASH').AttributeName);
```

Comment:
primaryHash is always required so I would move it out try/catch.

(also, what's with the weird 'if' statement?)